build-extras.gradle

```diff
@@ -0,0 +1,90 @@
+
+sourceCompatibility = 1.8
+targetCompatibility = 1.8
+
+ext {
+  commonsLang3Version = "3.8.1"
+  forgerockVersion = "5.5.1"
+  retrofit2Version = "2.5.0"
+  jacksonVersion = "2.9.7"
+  okHttpVersion = "3.12.0"
+}
+
+repositories {
+  google()
+  mavenCentral()
+  jcenter()
+  mavenLocal()
+  maven { url "https://trusona.jfrog.io/trusona/sdk-public-releases" }
+
+  maven {
+    credentials {
+      username = ARTIFACTORY_USERNAME
+      password = ARTIFACTORY_PASSWORD
+    }
+    url "https://trusona.jfrog.io/trusona/sdk-releases"
+  }
+  maven {
+    credentials {
+      username = FORGEROCK_MAVEN_USERNAME
+      password = FORGEROCK_MAVEN_PASSWORD
+    }
+    url "https://maven.forgerock.org/repo/openam-dependencies" 
+  }
+  maven { 
+    credentials {
+      username = FORGEROCK_MAVEN_USERNAME
+      password = FORGEROCK_MAVEN_PASSWORD
+    }
+    url "https://maven.forgerock.org/repo/private-releases" 
+  }
+  maven {
+    credentials {
+      username = FORGEROCK_MAVEN_USERNAME
+      password = FORGEROCK_MAVEN_PASSWORD
+    }
+   url "https://maven.forgerock.org/repo/releases" 
+  }
+}
+
+compileJava {
+  options.compilerArgs << "-Xlint:all"
+}
+
+test {
+  testLogging {
+    showStandardStreams = true
+    events "PASSED", "FAILED", "SKIPPED"
+  }
+
+  finalizedBy jacocoTestReport
+}
+
+configurations.all {
+  resolutionStrategy {
```

Comment:
What does this stuff do? Don't these just get pulled in automatically via the dependencies in build.gradle?

Reply:
did we actually run into that problem?

Reply:
yes - which is why i included the resolution strategy here. 
if we do not do this, there's no guarantee of which version of a dependency is picked at runtime.